source/particle/world.cc

```diff
@@ -627,7 +776,7 @@ namespace aspect
       // we create this derived class with an additional template. This makes it possible
       // to access the functionality through the base class, but create an object of this
       // derived class with the correct number of components at runtime.
-      template <int dim, int n_compositional_fields>
+      template <int dim>
       class SolutionEvaluatorsImplementation: public SolutionEvaluators<dim>
```

Comment:
If we do no longer need this template can we remove the distinction `SolutionEvaluators/SolutionEvaluatorsImplementation` again? That would simplify the code somewhat.

Reply:
Yes, this could go. Do you want to review this in a single large PR or separately?